SimPEG/electromagnetics/analytics/NSEM.py

```diff
@@ -15,14 +15,14 @@
 )
 
 # matrix P relating Up and Down components with E and H fields
-_P = lambda z: np.matrix([[1.0, 1,], [-1.0 / z, 1.0 / z],], dtype="complex_",)
-_Pinv = lambda z: np.matrix([[1.0, -z], [1.0, z]], dtype="complex_") / 2.0
+_P = lambda z: np.array([[1.0, 1,], [-1.0 / z, 1.0 / z],], dtype="complex_",)
+_Pinv = lambda z: np.array([[1.0, -z], [1.0, z]], dtype="complex_") / 2.0
 
 # matrix T for transition of Up and Down components accross a layer
-_T = lambda h, k: np.matrix(
+_T = lambda h, k: np.array(
     [[np.exp(1j * k * h), 0.0], [0.0, np.exp(-1j * k * h)]], dtype="complex_"
 )
-_Tinv = lambda h, k: np.matrix(
+_Tinv = lambda h, k: np.array(
     [[np.exp(-1j * k * h), 0.0], [0.0, np.exp(1j * k * h)]], dtype="complex_"
```

Comment:
same here (regarding explicit complex type)

Reply:
Good, I only looked into the matrix. Will remove these complex declarations.